src/stdlib_math.fypp

```diff
@@ -292,4 +298,27 @@ contains
     end function clip_${k1}$
 
     #:endfor
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    !> Determine the greatest common divisor of two integers
+    !> ([Specification](../page/specs/stdlib_math.html#gcd))
+    !>
+    !> Version: experimental
+    elemental function gcd_${k1}$(a, b) result(res)
+        ${t1}$, intent(in) :: a
+        ${t1}$, intent(in) :: b
+        ${t1}$ :: res
+
+        ${t1}$ :: rem, tmp
+
+        rem = min(abs(a), abs(b))
+        res = max(abs(a), abs(b))
+        do while (rem /= 0_${k1}$)
+          tmp = rem
+          rem = mod(res, rem)
+          res = tmp
+        end do
+    end function gcd_${k1}$
```

Comment:
If anyone ever wonder about performance in GCD there is a interesting benchmark
[here (in C)](https://hbfs.wordpress.com/2013/12/10/the-speed-of-gcd/). In conclusion it seems that this implementation is clear and fast enough.
PS: There is also the [gmplib's documentation](https://gmplib.org/manual/Greatest-Common-Divisor-Algorithms) that explores other GCD algorithms.

I believe this patch is good to go